webdriver/tests/bidi/browsing_context/locate_nodes/locator.py

```diff
@@ -0,0 +1,221 @@
+import pytest
+
+
+from ... import any_string, recursive_compare
+
+
+@pytest.mark.parametrize("type,value", [
+    ("css", "div"),
+    ("xpath", "//div"),
+    ("innerText", "foobarBARbaz")
+])
+@pytest.mark.asyncio
+async def test_find_by_locator(bidi_session, inline, top_context, type, value):
+    url = inline("""<div data-class="one">foobarBARbaz</div><div data-class="two">foobarBARbaz</div>""")
+    await bidi_session.browsing_context.navigate(
+        context=top_context["context"], url=url, wait="complete"
+    )
+
+    result = await bidi_session.browsing_context.locate_nodes(
+        context=top_context["context"],
+        locator={ "type": type, "value": value }
+    )
+
+    expected = [
+        {
+            "type": "node",
+            "sharedId": any_string,
+            "value": {
+                "attributes": {"data-class":"one"},
+                "childNodeCount": 1,
+                "children": [],
```

Comment:
I think children will be missing if empty

Reply:
Tell that to the [tests for serializing nodes from script](https://github.com/web-platform-tests/wpt/blob/master/webdriver/tests/bidi/script/evaluate/result_node.py), which appear to expect an empty `children` array. Might be a by-product of using `recursive_compare` for validation. I’ve no idea. But this follows the pattern set elsewhere, and absent evidence to the contrary, I’d be disinclined to change this.

Reply:
I believe those tests use max_dom_depth=1 and in this test max_dom_depth should 0 (the default). And node serialization steps #8 and 9 here https://w3c.github.io/webdriver-bidi/#serialize-as-a-remote-value say:

8) If serialization options["maxDomDepth"] is equal to 0, ...  let children be null.
9) If children is not null, set serialized["children"] to children.

Also, the children are not present in the actual serialization impl in Chromium. So it looks like this test either needs to add serializationOptions.maxDomDepth=1 or remove children? (or I am overlooks anything)

Reply:
That is a detail I apparently overlooked. I'll change the expected values.